### DISCUSSION TITLE
How to get Content-Length header without downloading the resource?

### DISCUSSION THREAD

**Discussion by enricorotundo** [go-resty/resty#565] [Q&A] [answered]
I'd need to prod a number of large files over HTTP to get their `Content-Length`. How to do it in resty?

I see there's a way to [do a HEAD request](https://github.com/go-resty/resty#sample-delete-head-options), however while curl works:

```
curl -L -I https://gist.githubusercontent.com/enricorotundo/990f0ad01a50d08dfb580e4ad404870e/raw/aa6934257351a0da93f1e740c72f27128590cebc/foo_data.txt
HTTP/1.1 200 OK
[...]
Content-Length: 574
[...]
```

the following resty code returns nothing:

```
response, err := client.R().EnableTrace().Head("https://gist.githubusercontent.com/enricorotundo/990f0ad01a50d08dfb580e4ad404870e/raw/aa6934257351a0da93f1e740c72f27128590cebc/foo_data.txt")
```

What am I missing here?

Ah snap! How about this? 👇 
https://github.com/go-resty/resty/blob/313f4190d9b8f7605ff298b769de613621c50220/response.go#L116-L118

**Comment by enricorotundo** [comment]
@jeevatkm Can you shed a light please? 🙃

**Comment by jeevatkm** [comment, marked as answer]
@enricorotundo I'm sorry for the delay. You could do -

```go
client := resty.New()
rsp, err := client.R().EnableTrace().Head("https://gist.githubusercontent.com/enricorotundo/990f0ad01a50d08dfb580e4ad404870e/raw/aa6934257351a0da93f1e740c72f27128590cebc/foo_data.txt")
fmt.Println(err)
fmt.Println(rsp.Header().Get("Content-Length"))
``` 
Output:
```
574
```
